File: app/src/main/java/com/example/techfix/data/dao/UserDao.java
package com.example.techfix.data.dao;
import androidx.room.*;
import com.example.techfix.data.entity.User;
@Dao public interface UserDao {
 @Insert(onConflict=OnConflictStrategy.ABORT) long insertUser(User user);
 @Query("SELECT * FROM users WHERE email = :email AND password = :password LIMIT 1") User login(String email,String password);
 @Query("SELECT * FROM users WHERE email = :email LIMIT 1") User getUserByEmail(String email);
 @Query("SELECT * FROM users WHERE userId = :id") User getUserById(int id);
}

File: app/src/main/java/com/example/techfix/data/dao/UserDao.kt
package com.example.techfix.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.techfix.data.entity.User

@Dao
interface UserDao {

    @Insert(onConflict = OnConflictStrategy.ABORT)
    suspend fun insertUser(user: User): Long

    @Query("SELECT * FROM users WHERE email = :email AND password = :password LIMIT 1")
    suspend fun login(email: String, password: String): User?

    @Query("SELECT * FROM users WHERE email = :email LIMIT 1")
    suspend fun getUserByEmail(email: String): User?

    @Query("SELECT * FROM users WHERE userId = :id")
    suspend fun getUserById(id: Int): User?
}